VmAgent.Core/Interfaces/DockerContainerEngine.cs

```diff
@@ -603,7 +603,22 @@ private IList<string> GetStartGameCmd(SessionHostsStartInfo request)
             {
                 if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
                 {
-                    return new List<string>() { $"cmd /c {request.StartGameCommand}" };
+                    var windowsStartCommand = new List<string>();
+                    
+                    if (request.WindowsCrashDumpConfiguration?.IsEnabled == true)
+                    {
+                        // set crash dump registry keys on startup
+                        windowsStartCommand.AddRange(new string[] {
+                            $"cmd /c reg add \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps\" /v DumpFolder /t REG_EXPAND_SZ /d %PF_SERVER_DUMP_DIRECTORY% /f ;",
+                            $"cmd /c reg add \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps\" /v DumpCount /t REG_DWORD /d 10 /f ;",
+                            $"cmd /c reg add \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps\" /v DumpType /t REG_DWORD /d {request.WindowsCrashDumpConfiguration.DumpType} /f ;",
+                            $"cmd /c reg add \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps\" /v CustomDumpFlags /t REG_DWORD /d {request.WindowsCrashDumpConfiguration.CustomDumpFlags} /f ;",
```

Comment:
Will the request values always be set? Should we have default? Is it ok if its null?

Reply:
Those are both ints, so as long as `WindowsCrashDumpConfiguration` isn't null (which we check for) then they will at least have a value. And while 0 is a value value for both of these registry keys so that's not an issue. I think validation at build creation time should be sufficient